examples/mcpwm-simple.rs

```diff
@@ -70,22 +70,24 @@ fn main() -> anyhow::Result<()> {
     println!("Configuring MCPWM");
 
     let peripherals = Peripherals::take().unwrap();
-    let config = OperatorConfig::default().frequency(25.kHz());
-    let mcpwm = Mcpwm::new(peripherals.mcpwm0.mcpwm)?;
-    let mut operator = Operator::new(
-        peripherals.mcpwm0.operator0,
-        &mcpwm,
-        &config,
-        peripherals.pins.gpio4,
-        peripherals.pins.gpio5,
-    )?;
+    let timer_config = TimerConfig::default().frequency(25.kHz());
+    let operator_config = OperatorConfig::default();
+    let timer = Mcpwm::new(peripherals.mcpwm0.timer, timer_config)?;
 
-    println!("Starting duty-cycle loop");
+    let timer = timer.into_connection()
+        .attatch_operator0(
+            peripherals.mcpwm0.operator0,
+            operator_config,
+            peripherals.pins.gpio4,
+            peripherals.pins.gpio5,
+        )?;
+
+    let (timer, operator, _, _) = timer.split();
```

Comment:
Here is one of the larger changes from "MCPWM from IDF <= 4.4" to "MCPWM from IDF 5.0". In 4.4, the concept of timer and operator is used interchangeably since every timer is hardcoded to exactly one operator(In #93 I expose the type `Operator` which control both the operator and its timer).

However, in 5.0 the mapping of timer to operator is no longer hardcoded. An operator can have any timer within the same MCPWM unit, thus multiple operators may share the same timer. To describe this, I have in this PR split this into two types `Operator` and `Timer`.

Since an operator needs a timer to work, we need to somehow ensure that the timer will live atleast as long as the operator. This could be done by having `Operator` hold a reference to its `Timer`. That would however prevent useful things like changing the frequency of the timer while an operator holds a reference to it.

(Please let me know if there is a better way/if this would not work)
To get around this problem I created `TimerConnection` which takes ownership of a timer and 0-3 operators. There is no way to move the `Operators` or the `Timer` out of the `TimerConnection` thus they are garantueed to live as the `Timer`. (mem::swap/replace would be one way to do it. However due to all members having unique types, it is impossible to find an instance of the same type to swap with.)

To get access to the objects within `TimerConnection` one has to call `TimerConnection::split()` which returns a tuple with mut references to every member.